Library/brew.rb

```diff
@@ -4,16 +4,17 @@
 
 HOMEBREW_BREW_FILE = ENV['HOMEBREW_BREW_FILE']
 
-if ARGV == %w{--prefix}
-  puts File.dirname(File.dirname(HOMEBREW_BREW_FILE))
-  exit 0
-end
-
 require 'pathname'
 HOMEBREW_LIBRARY_PATH = Pathname.new(__FILE__).realpath.parent.join("Homebrew")
 $:.unshift(HOMEBREW_LIBRARY_PATH.to_s)
+$:.unshift("#{HOMEBREW_LIBRARY_PATH}/test/lib") if ENV["HOMEBREW_TEST_LIB"]
```

Comment:
I kinda want to avoid injecting test code here, so there won't be case when `HOMEBREW_TEST_LIB` is accidentally set.

In fact we can inject test config when invoking brew command, like this: (If you don't think this it too hacky)
```ruby
  def cmd_output *args
    Utils.popen_read(RUBY_PATH,
                     "-W0",
                     "-I#{HOMEBREW_LIBRARY_PATH}/test/lib",
                     "-rpathname",
                     "-rconfig",
                     "#{(HOMEBREW_LIBRARY_PATH/"../brew.rb").resolved_path}",
                     *args).chomp
  end
```